bundles/core/org.eclipse.smarthome.core.audio/src/main/java/org/eclipse/smarthome/core/audio/AudioFormat.java

```diff
@@ -238,6 +240,132 @@ public boolean isCompatible(AudioFormat audioFormat) {
         return true;
     }
 
+    /**
+     * Determines the best match between a list of audio formats supported by a source and a sink.
+     *
+     * @param inputs the supported audio formats of an audio source
+     * @param outputs the supported audio formats of an audio sink
+     * @return the best matching format or null, if source and sink are incompatible
+     */
+    public static AudioFormat getBestMatch(Set<AudioFormat> inputs, Set<AudioFormat> outputs) {
+        AudioFormat preferredFormat = getPreferredFormat(inputs);
+        for (AudioFormat output : outputs) {
+            if (output.isCompatible(preferredFormat)) {
+                return preferredFormat;
+            } else {
+                for (AudioFormat input : inputs) {
+                    if (output.isCompatible(input)) {
+                        return input;
+                    }
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Gets the first concrete AudioFormat in the passed set or a preferred one
+     * based on 16bit, 16KHz, big endian default
+     *
+     * @param audioFormats The AudioFormats from which to choose
+     * @return The preferred AudioFormat. A passed concrete format is preferred adding
+     *         default values to an abstract AudioFormat in the passed Set.
```

Comment:
`in the passed Set` - okay, the data type is `Set`, but in the text, I assume we should use a lower case (but I am not an expert) :wink: